lib/elliptic/eddsa/index.js

```diff
@@ -0,0 +1,143 @@
+'use strict';
+
+var bn = require('bn.js');
+var hash = require('hash.js');
+var elliptic = require('../../elliptic');
+var utils = elliptic.utils;
+var assert = utils.assert;
+var KeyPair = require('./key');
+var Signature = require('./signature');
+
+function parseBytes(bytes) {
+  return typeof bytes === 'string' ? utils.toArray(bytes, 'hex') :
+                                     bytes;
+}
+
+function EDDSA(curve) {
+  if (!(this instanceof EDDSA))
+    return new EDDSA(curve);
+
+  assert(typeof curve === 'string');
+  assert(curve === 'ed25519', 'only tested with ed25519 so far');
```

Comment:
Yeah, just to be clear, and only taking string of ed25519 for the moment
easy wat of checking the only tested curve is used.